pkg/mcp/sink/client_sink.go

```diff
@@ -33,43 +34,39 @@ var (
 // Client implements the client for the MCP source service. The client is the
 // sink and receives configuration from the server.
 type Client struct {
-	// Client receives configuration using the ResourceSource RPC service
-	stream mcp.ResourceSource_EstablishResourceStreamClient
-
 	client mcp.ResourceSourceClient
 	*Sink
-	reporter monitoring.Reporter
 }
 
 // NewClient returns a new instance of Client.
 func NewClient(client mcp.ResourceSourceClient, options *Options) *Client {
 	return &Client{
-		Sink:     New(options),
-		reporter: options.Reporter,
-		client:   client,
+		Sink:   New(options),
+		client: client,
 	}
 }
 
 var reconnectTestProbe = func() {}
 
 func (c *Client) Run(ctx context.Context) {
-	// The first attempt is immediate.
-	retryDelay := time.Nanosecond
+	var err error
+	var stream Stream
 
 	for {
+		backoffPolicy := backoff.NewExponentialBackOff()
+		backoffPolicy.InitialInterval = time.Nanosecond
+		backoffPolicy.MaxElapsedTime = 0
+		t := backoff.NewTicker(backoffPolicy)
 		// connect w/retry
 		for {
 			select {
 			case <-ctx.Done():
 				return
-			case <-time.After(retryDelay):
+			case <-t.C:
```

Comment:
It wouldn't hurt to call `t.Stop()` here when the context is done. The ticker's finalizer calls Stop but that could happen arbitrarily in the future.